--- a/Makefile.in	1900-01-00 00:00:00 +0000
+++ b/Makefile.in	1900-01-00 00:00:00 +0000
@@ -323,7 +323,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.5 gnu
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = gnulib-local gettext-runtime libtextstyle gettext-tools
+SUBDIRS = $(VCPKG_GETTEXT_SUBDIRS)
 changelog_etc = \
   gettext-runtime/ChangeLog.0 \
   gettext-runtime/doc/ChangeLog.0 \
@@ -885,7 +885,7 @@
 #distdir: distdir1
 .PHONY: distdir1
 distdir1:
-	cd gettext-runtime/intl && $(MAKE) $(AM_MAKEFLAGS)
+	cd gettext-runtime/intl && $(MAKE) $(AM_MAKEFLAGS) libgnuintl.h
 	cd libtextstyle && $(MAKE) $(AM_MAKEFLAGS)
 
 # Generate ChangeLog.
