diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index 4132d60f79..c1e933437d 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -199,7 +199,7 @@ if(WIN32)
 endif()
 
 wx_get_install_dir(include)
-if(WIN32_MSVC_NAMING)
+if(FALSE)
     if(wxBUILD_SHARED)
         set(lib_suffix "_dll")
     else()
@@ -219,7 +219,7 @@ if(wxBUILD_CUSTOM_SETUP_HEADER_PATH)
     set(wxSETUP_HEADER_PATH ${wxBUILD_CUSTOM_SETUP_HEADER_PATH})
 else()
     # Set path where setup.h will be created
-    if(WIN32_MSVC_NAMING)
+    if(FALSE)
         set(lib_unicode u)
         set(wxSETUP_HEADER_PATH
             ${wxOUTPUT_DIR}/${wxPLATFORM_LIB_DIR}/${wxBUILD_TOOLKIT}${lib_unicode})
diff --git a/build/cmake/install.cmake b/build/cmake/install.cmake
index 88591a8c05..45b4effca0 100644
--- a/build/cmake/install.cmake
+++ b/build/cmake/install.cmake
@@ -57,18 +57,10 @@ else()
         PATTERN "protocols" EXCLUDE
     )
 
-    install(
-        FILES "${wxOUTPUT_DIR}/wx/config/${wxBUILD_FILE_ID}"
-        DESTINATION "${library_dir}/wx/config"
-        PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
-                    GROUP_EXECUTE GROUP_READ
-                    WORLD_EXECUTE WORLD_READ
-        )
-
     wx_get_install_platform_dir(runtime)
     install(DIRECTORY DESTINATION "${runtime_dir}")
     install(CODE "execute_process( \
-        COMMAND ${CMAKE_COMMAND} -E create_symlink \
+        COMMAND ${CMAKE_COMMAND} -E copy \
         \"${CMAKE_INSTALL_PREFIX}/${library_dir}/wx/config/${wxBUILD_FILE_ID}\" \
         \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${runtime_dir}/wx-config\" \
         )"
@@ -77,7 +69,7 @@ else()
 endif()
 
 wx_get_install_dir(library)
-set(wx_cmake_dir "${library_dir}/cmake/wxWidgets-${wxMAJOR_VERSION}.${wxMINOR_VERSION}")
+set(wx_cmake_dir "${library_dir}/cmake/wxWidgets")
 
 install(EXPORT wxWidgetsTargets NAMESPACE wx:: DESTINATION "${wx_cmake_dir}/${wxPLATFORM_LIB_DIR}")
 
