diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index e76dff6..eb4edc0 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -156,7 +156,7 @@ if(WIN32)
     endif()
 endif()
 
-if(WIN32_MSVC_NAMING)
+if(0)
     if(wxBUILD_SHARED)
         set(lib_suffix "_dll")
     else()
--- a/build/cmake/install.cmake	1900-01-00 00:00:00 +0000
+++ b/build/cmake/install.cmake	1900-01-00 00:00:00 +0000
@@ -53,17 +53,9 @@
         DIRECTORY "${wxSETUP_HEADER_PATH}"
         DESTINATION "lib/wx/include")
 
-    install(
-        FILES "${wxOUTPUT_DIR}/wx/config/${wxBUILD_FILE_ID}"
-        DESTINATION "lib/wx/config"
-        PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
-                    GROUP_EXECUTE GROUP_READ
-                    WORLD_EXECUTE WORLD_READ
-        )
-
     install(DIRECTORY DESTINATION "bin")
     install(CODE "execute_process( \
-        COMMAND ${CMAKE_COMMAND} -E create_symlink \
+        COMMAND ${CMAKE_COMMAND} -E copy \
         \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib/wx/config/${wxBUILD_FILE_ID}\" \
         \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/bin/wx-config\" \
         )"
