diff -Nur wxWidgets/include/wx/unix/apptrait.h wxWidgets-Qt-Android/include/wx/unix/apptrait.h
--- wxWidgets/include/wx/unix/apptrait.h	2026-07-24 10:53:33
+++ wxWidgets-Qt-Android/include/wx/unix/apptrait.h	2026-07-10 16:30:07
@@ -84,7 +84,7 @@
 #endif // wxUSE_SOCKETS
 
 #if wxUSE_EVENTLOOP_SOURCE
-    virtual wxEventLoopSourcesManagerBase* GetEventLoopSourcesManager() override;
+    virtual wxEventLoopSourcesManagerBase* GetEventLoopSourcesManager();
 #endif
 };
 
diff -Nur wxWidgets/src/common/intl.cpp wxWidgets-Qt-Android/src/common/intl.cpp
--- wxWidgets/src/common/intl.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/common/intl.cpp	2026-07-10 16:26:49
@@ -33,6 +33,7 @@
 #endif // WX_PRECOMP
 
 #include <locale.h>
+#include <langinfo.h>
 
 // standard headers
 #include <ctype.h>
diff -Nur wxWidgets/src/qt/accel.cpp wxWidgets-Qt-Android/src/qt/accel.cpp
--- wxWidgets/src/qt/accel.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/accel.cpp	2026-07-10 16:31:58
@@ -13,7 +13,8 @@
 #include "wx/accel.h"
 #include "wx/qt/private/converter.h"
 #include <QtCore/QVariant>
-#include <QShortcut>
+//#include <QShortcut>
+#include <QtGui/qshortcut.h>
 
 // ----------------------------------------------------------------------------
 // wxAccelRefData: the data used by wxAcceleratorTable
diff -Nur wxWidgets/src/qt/app.cpp wxWidgets-Qt-Android/src/qt/app.cpp
--- wxWidgets/src/qt/app.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/app.cpp	2026-07-10 16:32:30
@@ -15,7 +15,8 @@
 #include "wx/qt/private/converter.h"
 #include <QtCore/QStringList>
 #include <QtWidgets/QApplication>
-#include <QSurfaceFormat>
+//#include <QSurfaceFormat>
+#include <QtGui/qsurfaceformat.h>
 
 #include <algorithm>
 #include <vector>
diff -Nur wxWidgets/src/qt/dcprint.cpp wxWidgets-Qt-Android/src/qt/dcprint.cpp
--- wxWidgets/src/qt/dcprint.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/dcprint.cpp	2026-07-10 16:34:41
@@ -17,7 +17,8 @@
 
 #include <QtGui/QPainter>
 
-#include <QPrinter>
+//#include <QPrinter>
+#include <QtPrintSupport/qprinter.h>
 
 wxIMPLEMENT_ABSTRACT_CLASS(wxPrinterDCImpl, wxQtDCImpl)
 
diff -Nur wxWidgets/src/qt/evtloop.cpp wxWidgets-Qt-Android/src/qt/evtloop.cpp
--- wxWidgets/src/qt/evtloop.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/evtloop.cpp	2026-07-10 16:36:31
@@ -257,13 +257,15 @@
     }
 };
 
+#if 0
 wxEventLoopSourcesManagerBase* wxGUIAppTraits::GetEventLoopSourcesManager()
 {
     static wxQtEventLoopSourcesManager s_eventLoopSourcesManager;
     return &s_eventLoopSourcesManager;
 }
+#endif
 
-#if !wxUSE_CONSOLE_EVENTLOOP
+#if 0
 
 // Use the GUI event loop sources manager if console support is disabled
 // (needed by some common code, will raise an undefinied reference if not done)
diff -Nur wxWidgets/src/qt/graphics.cpp wxWidgets-Qt-Android/src/qt/graphics.cpp
--- wxWidgets/src/qt/graphics.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/graphics.cpp	2026-07-10 16:48:57
@@ -13,12 +13,12 @@
 
 #if wxUSE_GRAPHICS_CONTEXT
 
-#include <QApplication>
-#include <QBitmap>
-#include <QPainter>
-#include <QPainterPath>
-#include <QPicture>
-#include <QWidget>
+#include <QtWidgets/qapplication.h>
+#include <QtGui/QBitmap>
+#include <QtGui/QPainter>
+#include <QtGui/QPainterPath>
+#include <QtGui/QPicture>
+#include <QtWidgets/QWidget>
 
 #ifndef WX_PRECOMP
     #include "wx/bitmap.h"
diff -Nur wxWidgets/src/qt/menu.cpp wxWidgets-Qt-Android/src/qt/menu.cpp
--- wxWidgets/src/qt/menu.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/menu.cpp	2026-07-10 16:37:38
@@ -14,7 +14,7 @@
 #include "wx/qt/private/winevent.h"
 #include "wx/stockitem.h"
 
-#include <QActionGroup>
+#include <QtWidgets/qactiongroup.h>
 #include <QtWidgets/QMenu>
 #include <QtWidgets/QMenuBar>
 
diff -Nur wxWidgets/src/qt/menuitem.cpp wxWidgets-Qt-Android/src/qt/menuitem.cpp
--- wxWidgets/src/qt/menuitem.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/menuitem.cpp	2026-07-10 16:41:24
@@ -15,9 +15,9 @@
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/winevent.h"
 
-#include <QAction>
-#include <QMenuBar>
-#include <QRegularExpression>
+#include <QtWidgets/qaction.h>
+#include <QtWidgets/qmenubar.h>
+#include <QtGui/qvalidator.h>
 
 class wxQtAction : public QAction
 {
diff -Nur wxWidgets/src/qt/printdlg.cpp wxWidgets-Qt-Android/src/qt/printdlg.cpp
--- wxWidgets/src/qt/printdlg.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/printdlg.cpp	2026-07-10 16:43:31
@@ -17,11 +17,10 @@
 #include "wx/qt/printdlg.h"
 #include "wx/qt/private/converter.h"
 
-#include <QPrinter>
-#include <QPrinterInfo>
-#include <QPrintDialog>
-#include <QPageSetupDialog>
-#include <QPageSize>
+#include <QtPrintSupport/qprinter.h>
+#include <QtPrintSupport/qprinterinfo.h>
+#include <QtPrintSupport/qprintdialog.h>
+#include <QtPrintSupport/qpagesetupdialog.h>
 
 
 namespace // anonymous
diff -Nur wxWidgets/src/qt/printqt.cpp wxWidgets-Qt-Android/src/qt/printqt.cpp
--- wxWidgets/src/qt/printqt.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/printqt.cpp	2026-07-10 16:43:58
@@ -17,7 +17,7 @@
 #include "wx/qt/printdlg.h"
 #include "wx/private/print.h"
 
-#include <QPrinter>
+#include <QtPrintSupport/qprinter.h>
 
 wxIMPLEMENT_DYNAMIC_CLASS(wxQtPrinter, wxPrinterBase);
 
diff -Nur wxWidgets/src/qt/textentry.cpp wxWidgets-Qt-Android/src/qt/textentry.cpp
--- wxWidgets/src/qt/textentry.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/textentry.cpp	2026-07-10 16:45:06
@@ -15,7 +15,7 @@
 
 #include <QtCore/QStringListModel>
 #include <QtWidgets/QCompleter>
-#include <QFileSystemModel>
+#include <QtGui/QFileSystemModel>
 
 #include <QtWidgets/QComboBox>
 #include <QtWidgets/QLineEdit>
diff -Nur wxWidgets/src/qt/toolbar.cpp wxWidgets-Qt-Android/src/qt/toolbar.cpp
--- wxWidgets/src/qt/toolbar.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/toolbar.cpp	2026-07-10 16:45:32
@@ -11,7 +11,7 @@
 
 #if wxUSE_TOOLBAR
 
-#include <QActionGroup>
+#include <QtGui/QActionGroup>
 #include <QtWidgets/QToolButton>
 #include <QtWidgets/QToolBar>
 
diff -Nur wxWidgets/src/qt/window.cpp wxWidgets-Qt-Android/src/qt/window.cpp
--- wxWidgets/src/qt/window.cpp	2026-07-24 10:53:34
+++ wxWidgets-Qt-Android/src/qt/window.cpp	2026-07-10 16:45:55
@@ -19,7 +19,7 @@
 #include <QtWidgets/QScrollArea>
 #include <QtWidgets/QMainWindow>
 #include <QtWidgets/QMenu>
-#include <QShortcut>
+#include <QtGui/QShortcut>
 
 #ifndef WX_PRECOMP
     #include "wx/app.h"
