diff -Nur 4.1-138544d5b3.clean/source/CMakeLists.txt 4.1-138544d5b3.clean_patched/source/CMakeLists.txt
--- 4.1-138544d5b3.clean/source/CMakeLists.txt	2025-05-26 09:16:39.070794900 +0200
+++ 4.1-138544d5b3.clean_patched/source/CMakeLists.txt	2025-05-26 09:18:42.633498500 +0200
@@ -26,7 +26,7 @@
 mark_as_advanced(FPROFILE_USE FPROFILE_GENERATE NATIVE_BUILD)
 # X265_BUILD must be incremented each time the public API is changed
 set(X265_BUILD 215)
-configure_file("${PROJECT_SOURCE_DIR}/x265.def.in"
+configure_file("${PROJECT_SOURCE_DIR}/x265.def.in"
                "${PROJECT_BINARY_DIR}/x265.def")
 configure_file("${PROJECT_SOURCE_DIR}/x265_config.h.in"
                "${PROJECT_BINARY_DIR}/x265_config.h")
@@ -628,15 +628,7 @@
         list(APPEND PLATFORM_LIBS ${VLD_LIBRARIES})
         link_directories(${VLD_LIBRARY_DIRS})
     endif()
-    option(WINXP_SUPPORT "Make binaries compatible with Windows XP and Vista" OFF)
-    if(WINXP_SUPPORT)
-        # force use of workarounds for CONDITION_VARIABLE and atomic
-        # intrinsics introduced after XP
-        add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WINXP -D_WIN32_WINNT_WIN7=0x0601)
-    else(WINXP_SUPPORT)
-        # default to targeting Windows 7 for the NUMA APIs
-        add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WIN7)
-    endif(WINXP_SUPPORT)
+    add_definitions(-D_WIN32_WINNT=0x0501 -D_WIN32_WINNT_WIN7=0x0501)
 endif()
 
 if(POWER)
