diff -Nur x265-4.3/source/CMakeLists.txt x265-4.3-patched/source/CMakeLists.txt
--- x265-4.3/source/CMakeLists.txt	2026-07-31 13:04:17
+++ x265-4.3-patched/source/CMakeLists.txt	2026-09-03 16:45:09
@@ -836,15 +836,7 @@
         list(APPEND PLATFORM_LIBS ${VLD_LIBRARIES})
         link_directories(${VLD_LIBRARY_DIRS})
     endif()
-    option(WINXP_SUPPORT "Make binaries compatible with Windows XP and Vista" OFF)
-    if(WINXP_SUPPORT)
-        # force use of workarounds for CONDITION_VARIABLE and atomic
-        # intrinsics introduced after XP
-        add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WINXP -D_WIN32_WINNT_WIN7=0x0601)
-    else(WINXP_SUPPORT)
-        # default to targeting Windows 7 for the NUMA APIs
-        add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WIN7)
-    endif(WINXP_SUPPORT)
+    add_definitions(-D_WIN32_WINNT=0x0501 -D_WIN32_WINNT_WIN7=0x0501)
 endif()
 
 if(POWER)
