diff -Nur 8bc98c3a0d-84009aba94.clean/config.sub 8bc98c3a0d-84009aba94.clean-patched/config.sub
--- 8bc98c3a0d-84009aba94.clean/config.sub	2023-10-01 16:28:26
+++ 8bc98c3a0d-84009aba94.clean-patched/config.sub	2026-07-31 13:44:26
@@ -316,7 +316,7 @@
 	| tahoe | tic4x | tic54x | tic55x | tic6x | tic80 | tron \
 	| ubicom32 \
 	| v850 | v850e | v850e1 | v850e2 | v850es | v850e2v3 \
-	| we32k \
+	| we32k | riscv64 \
 	| x86 | xc16x | xstormy16 | xtensa \
 	| z8k | z80)
 		basic_machine=$basic_machine-unknown
