diff -Nur FFmpeg-n9.0/configure FFmpeg-n9.0-no-bcrypt/configure
--- FFmpeg-n9.0/configure	2026-08-03 21:52:07
+++ FFmpeg-n9.0-no-bcrypt/configure	2026-08-13 16:01:19
@@ -2611,7 +2611,6 @@
 "
 
 SYSTEM_LIBRARIES="
-    bcrypt
     vaapi_drm
     vaapi_x11
     vaapi_win32
@@ -4362,7 +4361,7 @@
 avfilter_suggest="libm stdatomic zlib"
 avformat_deps="avcodec avutil"
 avformat_suggest="libm network zlib stdatomic"
-avutil_suggest="amf clock_gettime ffnvcodec gcrypt libm zlib libdrm libmfx opencl openssl user32 vaapi vulkan videotoolbox corefoundation corevideo coremedia bcrypt stdatomic"
+avutil_suggest="amf clock_gettime ffnvcodec gcrypt libm zlib libdrm libmfx opencl openssl user32 vaapi vulkan videotoolbox corefoundation corevideo coremedia stdatomic"
 swresample_deps="avutil"
 swresample_suggest="libm libsoxr stdatomic"
 swscale_deps="avutil"
@@ -7097,8 +7096,6 @@
 done
 
 check_lib advapi32 "windows.h"            RegCloseKey          -ladvapi32
-check_lib bcrypt   "windows.h bcrypt.h"   BCryptGenRandom      -lbcrypt &&
-    check_cpp_condition bcrypt bcrypt.h "defined BCRYPT_RNG_ALGORITHM"
 check_lib ole32    "windows.h objbase.h"  CoTaskMemFree        -lole32
 check_lib shell32  "windows.h shellapi.h" CommandLineToArgvW   -lshell32
 check_lib psapi    "windows.h psapi.h"    GetProcessMemoryInfo -lpsapi
diff -Nur FFmpeg-n9.0/libavutil/random_seed.c FFmpeg-n9.0-no-bcrypt/libavutil/random_seed.c
--- FFmpeg-n9.0/libavutil/random_seed.c	2026-08-03 21:52:07
+++ FFmpeg-n9.0-no-bcrypt/libavutil/random_seed.c	2026-08-13 16:01:54
@@ -26,10 +26,6 @@
 #if HAVE_IO_H
 #include <io.h>
 #endif
-#if HAVE_BCRYPT
-#include <windows.h>
-#include <bcrypt.h>
-#endif
 #if CONFIG_GCRYPT
 #include <gcrypt.h>
 #elif CONFIG_OPENSSL
@@ -159,18 +155,6 @@
 int av_random_bytes(uint8_t* buf, size_t len)
 {
     int err;
-
-#if HAVE_BCRYPT
-    BCRYPT_ALG_HANDLE algo_handle;
-    NTSTATUS ret = BCryptOpenAlgorithmProvider(&algo_handle, BCRYPT_RNG_ALGORITHM,
-                                               MS_PRIMITIVE_PROVIDER, 0);
-    if (BCRYPT_SUCCESS(ret)) {
-        NTSTATUS ret = BCryptGenRandom(algo_handle, (PUCHAR)buf, len, 0);
-        BCryptCloseAlgorithmProvider(algo_handle, 0);
-        if (BCRYPT_SUCCESS(ret))
-            return 0;
-    }
-#endif
 
 #if HAVE_ARC4RANDOM_BUF
     arc4random_buf(buf, len);
